src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -889,10 +892,12 @@ protected virtual bool TryLoadFromCache(JsonOperationContext context)
 
                 SetRequestHeaders(sessionInfo, cachedChangeVector, request);
 
+                OnBeforeRequest?.Invoke(this, request);
                 var response = await SendRequestToServer(chosenNode, nodeIndex, context, command, shouldRetry, sessionInfo, request, url, token).ConfigureAwait(false);
                 if (response == null) // the fail-over mechanism took care of this
```

Comment:
What about failures here? How would you deal with the correlation?

Reply:
If the response cannot be null so the user can get it from the response, no?